hydra/_internal/hydra.py

```diff
@@ -156,16 +157,19 @@ def _get_cfg(
         self,
         config_name: Optional[str],
         overrides: List[str],
-        cfg_type: str,
         with_log_configuration: bool,
     ) -> DictConfig:
-        assert cfg_type in ["job", "hydra", "all"]
         cfg = self.compose_config(
             config_name=config_name,
             overrides=overrides,
             run_mode=RunMode.RUN,
             with_log_configuration=with_log_configuration,
         )
+        HydraConfig.instance().set_config(cfg)
```

Comment:
I would prefer not to introduce a side effect to a getter function.
Can the caller do this instead?

Reply:
I inlined the `_get_cfg` method in https://github.com/facebookresearch/hydra/pull/1696/commits/6828fcc003618a8a4dafd5a01ed3653fdd93655d.